drivers/serial/serial_esp32_usb.c

```diff
@@ -53,14 +58,17 @@ static void serial_esp32_usb_poll_out(const struct device *dev, unsigned char c)
 	ARG_UNUSED(dev);
 
 	/* Wait for space in FIFO */
-	while (usb_serial_jtag_ll_txfifo_writable() == 0) {
-		;
+	while (!usb_serial_jtag_ll_txfifo_writable() &&
+		s_usbserial_timeout < (USBSERIAL_TIMEOUT_MAX_US / 100)) {
+		esp_rom_delay_us(100);
```

Comment:
Would it make sense to use one of the Zephyr APIs for the delay here?

Reply:
done!